src/scripts/core/models/grid.ts

```diff
@@ -36,41 +32,73 @@ export class Grid extends Container {
 
     }
 
-    private clickCheck(tokenX: number, tokenY: number): void {
+    private clickCheck(targetToken: Token): void {
         //on first click, despite being defined in the constructor, activeToken appears undefined
         if(this.activeToken === undefined) {
             this.activeToken = 0;
         }
 
-        const location = [tokenX, tokenY];
-        this.matchCheck(location);
-
-        // if(this.activeToken === 0) {
-        //     this.activeToken = 1; 
-        //     eventEmitter.emit('tokenFirstClicked', tokenX, tokenY);
-        //     console.log("FirstClick", this.activeToken)
-        //     return;
-        // }
-
-        // if(this.activeToken === 1) {
-        //     this.activeToken = 2; 
-        //     eventEmitter.emit('tokenSecondClicked', tokenX, tokenY); 
-        //     console.log("SecondClick", this.activeToken)
-        //     return; 
-        // }
-    }
+        //on FirstClick
+        if((this.activeToken === 0) && (this.firstTokenLocation[0] === -1) && (this.firstTokenLocation[0] === -1)) {
+            targetToken.highLight();
+            this.activeToken = 1;
+            this.firstTokenLocation = targetToken.getLocation();
```

Comment:
I wonder, do you need to base this on the token location? By identifying the token within the array, you should be able to store which tokens are highlighted right?

Which also means your calculations below may be simplified